services/test_results.py

```diff
@@ -200,14 +200,23 @@ def build_message(self, url, test_instances):
         ]
 
         message += details
-
         failure_table = [
             "| {0} | <pre>{1}</pre> |".format(
-                self.insert_breaks(failed_test_name),
+                (
+                    "<br>".join(
+                        self.insert_breaks(
+                            f"{test_name}[{','.join(sorted(test_env_list))}]"
+                        )
+                        for test_name, test_env_list in sorted(
+                            failed_test_to_env_list.items(),
+                            key=lambda failure: failure[0],
+                        )
+                    )
+                ),
                 failure_message.replace("\n", "<br>"),
             )
-            for failed_test_name, failure_message in sorted(
-                failures.items(), key=lambda failure: failure[0]
+            for failure_message, failed_test_to_env_list in sorted(
+                failures.items(), key=lambda x: x[0]
```

Comment:
Might be missing something but couldn't you use `failures.keys()` and `x` instead of x[0]? same for the loop above

Reply:
do you mean to do something like: 
```
self.insert_breaks(
  f"{test_name}[{','.join(sorted(failed_test_to_env_list[test_name]))}]"
)
for test_name in sorted(
  failed_test_to_env_list.keys(),
)

Reply:
Rereading this now, your approach is fine, I was under the impression you only needed the keys rather than the key/value, so I was suggesting to use only the keys to make it more readable